webknossos/webknossos/client/__init__.py

```diff
@@ -0,0 +1,37 @@
+import os
+from functools import lru_cache
+
+from dotenv import load_dotenv
+from rich.prompt import Prompt
+
+from .defaults import DEFAULT_WEBKNOSSOS_URL
+from .generated import Client as GeneratedClient
+
+load_dotenv()
+
+
+@lru_cache(maxsize=None)
+def _get_token(webknossos_url: str) -> str:
+    if "WK_TOKEN" in os.environ:
+        return os.environ["WK_TOKEN"]
+    else:
+        # TODO  pylint: disable=fixme
+        # -validate token and ask again if necessary
+        # -ask if the token should be saved in some .env file
+        return Prompt.ask(
+            f"\nPlease enter your webknossos token as shown on {webknossos_url}/auth/token ",
+            password=True,
+        )
+
+
+@lru_cache(maxsize=None)
+def _get_generated_client(
+    webknossos_url: str = DEFAULT_WEBKNOSSOS_URL, with_token: bool = False
```

Comment:
What do you think about using the token by default if it's in the ENV? The parameter could be `enforce_token: bool = False` with the semantic that the user prompt is only shown if `enforce_token` is true and there is no WK_TOKEN in the env.. No strong opinion, though.

Reply:
I think that's exactly what `with_token` is doing, do you agree? But `enforce_token` or `needs_token` is a much better name, I'll adapt it. I'd be happy to discuss this further in a broader sense anyway (e.g. should we do user-wide caching?)

Reply:
> I think that's exactly what with_token is doing, do you agree?

I might be missing something, but currently `_get_generated_client()` will not use any tokens from ENV even if these exist. Instead, I have to explicitly pass `with_token=True`. My suggestion is to always use an existing token. If no token exists **and** `enforce_token=True`, the prompt will come up. Does this make sense?

> I'd be happy to discuss this further in a broader sense anyway (e.g. should we do user-wide caching?)

User-wide caching sounds like a nice-to-have. However, probably not high-pri for now and also we should communicate this clearly with the user so that tokens are not written to disk without their knowledge.

Reply:
Good point, I adapted it in the newest commit